xds/internal/xdsclient/tests/ads_stream_ack_nack_test.go

```diff
@@ -442,39 +453,32 @@ func (s) TestADS_ACK_NACK_ResourceIsNotRequestedAnymore(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Cancel the watch on the listener resource. This should result in a
-	// discovery request with no resource names.
+	// Cancel the watch on the listener resource. This should result in the
+	// existing connection to be management server getting closed.
 	ldsCancel()
-
-	// Verify that the discovery request matches expectation.
-	r, err = streamRequestCh.Receive(ctx)
-	if err != nil {
-		t.Fatal("Timeout when waiting for discovery request")
-	}
-	gotReq = r.(*v3discoverypb.DiscoveryRequest)
-	wantReq.ResourceNames = nil
-	if diff := cmp.Diff(gotReq, wantReq, protocmp.Transform()); diff != "" {
-		t.Fatalf("Unexpected diff in received discovery request, diff (-got, +want):\n%s", diff)
+	if _, err := conn.CloseCh.Receive(ctx); err != nil {
+		t.Fatalf("Timeout when expecting existing connection to be closed: %v", err)
 	}
+	streamRequestCh.Drain()
```

Comment:
why do we have to drain? May be good to put a comment to explain?

Reply:
Added a comment, and slightly re-worked the test since I saw another flake on Github actions.

Reply:
> we drain the request channel here so that if an empty discovery request was received, it is pulled out of the request channel

Based on the PR description, i thought this should never happen anymore?